### ISSUE TITLE
Rx Observable binding possible?

### ISSUE THREAD

**Issue by ken-okabe** [fable-compiler/Fable.Solid#2]
I really like this project because it's simply a wrapper of the cutting-edge web JS library and directly depends on the latest release of [solid.js v1.5](https://github.com/solidjs/solid/releases/tag/v1.5.0).

I have not used Solid seriously yet, but I recognized the framework is the fastest next to vanilla.js without virtual-DOM and way much simpler than React although taking advantage of JSX.

I also feel very promissing to have found the strong [ecosystem with user Add-on
](https://www.solidjs.com/ecosystem), so if something is really needed from the Fable side, we can take advantage of developing Fable Add-on.

I see several add-on for state management that is so flexible for them to allow unlike React, but I also see their native API has 

https://www.solidjs.com/docs/latest/api#observable

---------------

### observable

```js
import { observable } from "solid-js";

function observable<T>(input: () => T): Observable<T>;
```

This method takes a signal and produces an Observable. You can consume it from another Observable library of your choice, typically with the from operator.

```js
// How to integrate rxjs with a solid.js signal
import { observable } from "solid-js";
import { from } from "rxjs";

const [s, set] = createSignal(0);

const obsv$ = from(observable(s));

obsv$.subscribe((v) => console.log(v));
```

You can also use from without rxjs; see below.

### from
New in v1.1.0

```js
import { from } from "solid-js";

function from<T>(
  producer:
    | ((setter: (v: T) => T) => () => void)
    | {
        subscribe: (
          fn: (v: T) => void
        ) => (() => void) | { unsubscribe: () => void };
      }
): () => T;
```

A helper to make it easier to interop with external producers like RxJS observables or with Svelte Stores. This basically turns any subscribable (object with a subscribe method) into a Signal and manages subscription and disposal.

```js
const signal = from(obsv$);
```

It can also take a custom producer function where the function is passed a setter function returns a unsubscribe function:

```js
const clock = from((set) => {
  const t = setInterval(() => set(1), 1000);
  return () => clearInterval(t);
});
```
-----------------

So, currently, I wonder how it's possible to combine Fable.Solid and [Observable](http://fsprojects.github.io/FSharp.Control.Reactive/reference/fsharp-control-reactive-observablemodule.html) of Rx for F# that is
https://github.com/fsprojects/FSharp.Control.Reactive

In short, it would be very nice to have  [Observable](http://fsprojects.github.io/FSharp.Control.Reactive/reference/fsharp-control-reactive-observablemodule.html) of Rx version for the typical counter-demo.

```fsharp
[<JSX.Component>]
let Counter() =
    let count, setCount = Solid.createSignal(0)
    let doubled() = count() * 2
    let quadrupled() = doubled() * 2

    Html.fragment [
        Html.p $"{count()} * 2 = {doubled()}"
        Html.p $"{doubled()} * 2 = {quadrupled()}"
        Html.br []
        Html.button [
            Attr.className "button"
            Ev.onClick(fun _ -> count() + 1 |> setCount)
            Html.children [
                Html.text $"Click me!"
            ]
        ]
    ]
```

Thanks!

**Comment by alfonsogarciacaro** [comment]
I just realized that Github doesn't activate notifications for new repos in an organization, so I had missed your issue until now. Really sorry for that @stken2050 😞 

I didn't notice that Solid had an Observable API either, thanks for pointing that out. It would be possible to create bindings for RxJS (although that would have to come from a contribution), but Fable already supports the `Observable` module from FSharp.Core which is not as comprehensive as RxJS but already covers many use cases (mapping, merging...) and can be easily extended to support others (like throttling). Also, by using the built-in Observable you avoid having to bring in RxJS to your app, which is a big dependency.

We already explored using observables with [Fable.Store](https://github.com/fable-compiler/Fable.Store) with Svelte, Sutil and React, but it didn't catch on within the community. I think the reason is that at the end there are no many situations where the UI can benefit from dealing with observables as streams.

Anyways, it should be easy to write adapters so we can easily convert between F# Observables and Solid signals back and forth.

**Comment by ken-okabe** [comment]
@alfonsogarciacaro Thanks for your thought 🙂
Actually, so far I have developed some projects for myself with solid, and found the build-in mechanism of state management of solid works pretty well and concisely, and bridging every stream to other frameworks is somewhat redundant.
At the same time, thanks to my commitments,  I also found it's pretty easy to `map` from any Observable to solid stream, so if I really need I could build by myself.
Thanks again!
